packages/runtime/container-runtime/src/dataStores.ts

```diff
@@ -97,6 +99,10 @@ export class DataStores implements IDisposable {
         readonly referencedDataStoreCount: number;
     };
 
+    // Stores the ids of new data stores between two GC runs. This is used to notify the garbage collector of new
+    // root data stores that are added.
+    private dataStoresSinceLastGC: string[] = [];
```

Comment:
Can we somehow make it more generic? I.e., support future deeper paths, including DDSs and nested data stores (unified channels)? It would be impossible to find all the places like that when we implement channels...

Reply:
This layer is concerned about data stores and it can only detect attach messages for data stores. So, I believe this should be fine. If we make the API `rootDataStoreAdded` below generic, then this is just an implementation detail.